test/dependency-manager/spec/partials/spec-test-harness.ts

```diff
@@ -38,17 +42,44 @@ const recursiveMergeTree = (tree: RecursivePartialsTree): object[] => {
       const child_partials = recursiveMergeTree(child_tree);
 
       const merged_partials = tree.partials.map(current_partial => {
-        return child_partials.map(child_partial => {
+        let accums: object[] = [];
+        for (const child_partial of child_partials) {
           if (child_key.startsWith('_')) {
             let accum = current_partial;
-            for (const current_key of Object.keys(accum)) {
-              accum[current_key] = deepmerge(accum[current_key], { [child_key.replace('_', '')]: child_partial });
+            const resource_name = Object.keys(accum)[0];
```

Comment:
Can you explain what is going on here?

Reply:
If we didn't change what we're doing here, a bunch of tests where these partial specs are used would fail validation because we validate that we're not using `build` with `image` now